watcher/watcher.go

```diff
@@ -31,6 +34,87 @@ type WatchHandler struct {
 	currentPodListResourceVersion     string // current PodList version, used by watcher (https://kubernetes.io/docs/reference/using-api/api-concepts/#efficient-detection-of-changes)
 }
 
+// remove unused imageIDs and instanceIDs from storage. Update internal maps
+func (wh *WatchHandler) cleanUp(ctx context.Context) error {
+	storageImageIDs, err := wh.ListImageIDsFromStorage()
+	if err != nil {
+		logger.L().Ctx(ctx).Error("error to ListImageIDsFromStorage", helpers.Error(err))
+	}
+
+	storageInstanceIDs, err := wh.ListInstanceIDsFromStorage()
+	if err != nil {
+		logger.L().Ctx(ctx).Error("error to ListInstanceIDsFromStorage", helpers.Error(err))
+	}
+
+	// list Pods, extract their imageIDs and instanceIDs
+	podsList, err := wh.k8sAPI.ListPods("", map[string]string{})
+	if err != nil {
+		return err
+	}
+
+	// reset maps
+	wh.cleanUpMaps()
```

Comment:
[Question]: Why did you call the function `cleanUpMaps` and then add a comment saying "reset maps", why not name the function `resetMaps` or something similar?
Personally, I think such a function should not be called based on the data structure. Meaning, the data structure is of a type map (dictionary) that is representing the unique imagesIDs and instancesIDs, so the function can be called `resetIDs` or even `resetIDsStructures`. Think about it...

Reply:
Regarding your question, the `cleanUpMaps` does what its name states - cleans up the maps. The comment is related to the flow, meaning we clean the maps then build them again, that's why I wrote reset maps. I will try to make it more clear in the comments. 
I will also rename the function